otherlibs/stdune/src/path.ml

```diff
@@ -1130,7 +1130,19 @@ let is_directory t =
 
 let rmdir t = Unix.rmdir (to_string t)
 let unlink t = Fpath.unlink (to_string t)
-let link x y = Unix.link (to_string x) (to_string y)
+
+let link x y =
+  match Unix.link (to_string x) (to_string y) with
+  | exception Unix.Unix_error (Unix.EUNKNOWNERR -1142, syscall, arg)
+  (* Needed for OCaml < 5.1 *) ->
+    (* If we can't make a new hard link because we reached the limit on the
+       number of hard links per file, we fall back to copying. We expect that
```

Comment:
The bit about falling back to copying is out of place now.

Reply:
It's back. Accidentally deleted it.